packages/server/src/asyncQuestion/asyncQuestion.controller.ts

```diff
@@ -44,6 +45,62 @@ export class asyncQuestionController {
     private mailService: MailService,
   ) {}
 
+  @Post('comment')
+  @Roles(Role.STUDENT, Role.TA, Role.PROFESSOR)
+  async replyToQuestion(
+    @Body() body: any,
+    @User() user: UserModel,
+    @Res() res: Response,
+  ): Promise<Response> {
+    const { questionId, commentText } = body;
+    const question = await AsyncQuestionModel.findOne({
+      where: { id: questionId },
+      relations: ['course', 'creator'],
+    });
+
+    if (!question) {
+      res
+        .status(HttpStatus.NOT_FOUND)
+        .send({ message: ERROR_MESSAGES.questionController.notFound });
+      return;
+    }
+
+    // check if user has commented within 2 minutes
+    const lastComment = await AsyncQuestionCommentsModel.findOne({
+      where: { creator: user, question },
+      order: { createdAt: 'DESC' },
+    });
+
+    if (lastComment) {
+      const timeDiff = new Date().getTime() - lastComment.createdAt.getTime();
+      if (timeDiff < 30000) {
+        res.status(HttpStatus.BAD_REQUEST).send({
+          message: ERROR_MESSAGES.questionController.createQuestion.tooFast,
```

Comment:
Also, I realised it would be pretty easy to change the error message to be custom. Something like:
```suggestion
          message: `You are posting too quickly. Please try again in ${timeDiff / 60000}s`,
```